app/controllers/catkeys_controller.rb

```diff
@@ -20,18 +49,37 @@ def edit
   def update
     return unless enforce_versioning
 
-    form = CatkeyForm.new(@cocina)
-    form.validate(catkey: update_params[:catkey].strip)
-    form.save
-    Argo::Indexer.reindex_druid_remotely(@cocina.externalIdentifier)
-
-    msg = "Catkey for #{@cocina.externalIdentifier} has been updated!"
-    redirect_to solr_document_path(@cocina.externalIdentifier), notice: msg
+    @form = catkey_form
+    respond_to do |format|
+      if @form.validate(params[:catkey]) && @form.save
+        Argo::Indexer.reindex_druid_remotely(@cocina.externalIdentifier)
+        msg = "Catkeys for #{@cocina.externalIdentifier} have been updated!"
+        format.html { redirect_to solr_document_path(@cocina.externalIdentifier, format: :html), notice: msg }
+      else
+        format.turbo_stream do
+          render turbo_stream: turbo_stream.replace('modal-frame', partial: 'edit'), status: :unprocessable_entity
+        end
+      end
+    end
   end
 
   private
 
-  def update_params
-    params[CatkeyForm.model_name.param_key]
+  def catkey_form
+    # fetch catkeys from object
+    object_catkeys = @cocina.identification.catalogLinks.filter_map { |catalog_link| catalog_link if catalog_link.catalog == Constants::SYMPHONY }
+
+    # if there are no object catkeys, provide an initial blank row, else form is initialized with catkeys in the object
+    catkeys = if object_catkeys.size.zero?
+                [CatkeyRow.new(value: '', refresh: true)]
```

Comment:
This should be done via prepopulate!

Reply:
moved the blank catkey row logic to a prepopulator on the form